src/modules/frost/frost.md

```diff
@@ -0,0 +1,93 @@
+3. After receiving a share and commitment set from each participant, call
+   `secp256k1_frost_share_agg` to compute the aggregate share and group public
+   key. If this function returns an error, `secp256k1_frost_share_verify` is
+   called on each share to determine which participants submitted faulty
+   shares.
+4. Optionally compute the public verification share by calling
+   `secp256k1_frost_compute_pubshare` with the x-only public key of each
+   participant. This share is required to verify partial signatures generated
+   by `secp256k1_frost_partial_sig_verify`.
+5. Sign the `vss_hash` generated by `secp256k1_frost_share_agg` and distribute
+   the signature to each participant.
+6. Verify the signature received by each participant is a valid signature for
+   the `vss_hash` generated by `secp256k1_frost_share_agg`.
+
+# Tweaking
+
+
+A (Taproot) tweak can be added to the resulting public key with
+`secp256k1_xonly_pubkey_tweak_add`, after converting it to an xonly pubkey if
+necessary with `secp256k1_xonly_pubkey_from_pubkey`.
+
+An ordinary tweak can be added to the resulting public key with
+`secp256k1_ec_pubkey_tweak_add`, after converting it to an ordinary pubkey if
+necessary with `secp256k1_frost_pubkey_get`.
+
+Tweaks can also be chained together by tweaking an already tweaked key.
+
+# Signing
+
+1. Generate a keypair with `secp256k1_keypair_create` and obtain the xonly public key with `secp256k1_keypair_xonly_pub`.
+2. Optionally add a tweak by calling `secp256k1_frost_pubkey_tweak` and then `secp256k1_frost_pubkey_xonly_tweak_add` for a Taproot tweak and `secp256k1_frost_pubkey_ec_tweak_add` for an ordinary tweak.
+3. Generate a pair of secret and public nonce with `secp256k1_frost_nonce_gen` and send the public nonce to the other signers.
+4. Process the aggregate nonce with `secp256k1_frost_nonce_process`.
+5. Create a partial signature with `secp256k1_frost_partial_sign`.
+8. Verify the partial signatures (optional in some scenarios) with `secp256k1_frost_partial_sig_verify`.
+9. Someone (not necessarily the signer) obtains all partial signatures and aggregates them into the final Schnorr signature using `secp256k1_frost_partial_sig_agg`.
+
+The aggregate signature can be verified with `secp256k1_schnorrsig_verify`.
+
+Note that steps 1 to 4 can happen before the message to be signed is known to
+the signers. Therefore, the communication round to exchange nonces can be
+viewed as a pre-processing step that is run whenever convenient to the signers.
+This disables some of the defense-in-depth measures that may protect against
+API misuse in some cases. Similarly, the API supports an alternative protocol
+flow where generating the key (see Key Generation above) is allowed to happen
+after exchanging nonces (step 3).
+
+# Verification
+
+A participant who wants to verify the partial signatures, but does not sign itself may do so using the above instructions except that the verifier skips steps 1, 3 and 5.
```

Comment:
If you would like to document the atomic swap variant for FROST (assuming there is no additional assumption or round): https://github.com/ariard/secp256k1-zkp/commit/0180bfb045b2eb6e9b2d8036236253249766c220